src/test/java/com/alvarium/annotators/VulnerabilityAnnotatorTest.java

```diff
@@ -0,0 +1,129 @@
+ *******************************************************************************/
+package com.alvarium.annotators;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
+import com.alvarium.SdkInfo;
+import com.alvarium.contracts.Annotation;
+import com.alvarium.contracts.AnnotationType;
+import com.alvarium.hash.HashInfo;
+import com.alvarium.hash.HashType;
+import com.alvarium.sign.KeyInfo;
+import com.alvarium.sign.SignType;
+import com.alvarium.sign.SignatureInfo;
+import com.alvarium.utils.ImmutablePropertyBag;
+import com.alvarium.utils.PropertyBag;
+
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+public class VulnerabilityAnnotatorTest {
+
+    @Rule
+    public TemporaryFolder dir = new TemporaryFolder();
+
+    public void executeShouldReturnAnnotation() throws AnnotatorException, IOException {
+    // construct annotator
+    final AnnotatorFactory annotatorFactory = new AnnotatorFactory();
+    final KeyInfo pubKey = new KeyInfo("./src/test/java/com/alvarium/annotators/public.key", 
+        SignType.Ed25519);
+    final KeyInfo privKey = new KeyInfo("./src/test/java/com/alvarium/annotators/private.key",
+        SignType.Ed25519);
+    final SignatureInfo sigInfo = new SignatureInfo(pubKey, privKey);
+    final AnnotationType[] annotators = { AnnotationType.VULNERABILITY};
+    final SdkInfo config = new SdkInfo(annotators, new HashInfo(HashType.SHA256Hash), sigInfo, null);
+
+    final Annotator annotator = annotatorFactory.getAnnotator(AnnotationType.VULNERABILITY, config);
+    
+    //Writing pom.xml file with a dependency that has a known vulnerability
+    //Therefore a non satisfied annotation should be returned
+    final File pomXmlFile  = dir.newFile("pom.xml");
+    try (FileWriter writer = new FileWriter(pomXmlFile)) {
+    writer.write("<project>\n");
+    writer.write("  <modelVersion>4.0.0</modelVersion>\n");
+    writer.write("  <groupId>com.example</groupId>\n");
+    writer.write("  <artifactId>my-project</artifactId>\n");
+    writer.write("  <version>1.0.0</version>\n");
+    writer.write("\n");
+    writer.write("  <dependencies>\n");
+    writer.write("    <dependency>\n");
+    writer.write("      <groupId>junit</groupId>\n");
+    writer.write("      <artifactId>junit</artifactId>\n");
+    writer.write("      <version>4.12</version>\n");
+    writer.write("      <scope>test</scope>\n");
+    writer.write("    </dependency>\n");
+    writer.write("  </dependencies>\n");
+    writer.write("</project>\n");
+    }
+
+    // dummy data and empty prop bag
+    final byte[] data = "test data".getBytes();
+    HashMap<String, Object> map = new HashMap<String, Object>();
+    map.put(AnnotationType.VULNERABILITY.name(), pomXmlFile.toPath().getParent().toString());
+    final PropertyBag ctx = new ImmutablePropertyBag(map);
+
+    final Annotation annotation = annotator.execute(ctx, data);
+
+    assert !annotation.getIsSatisfied();

```

Comment:
Suggest deleting pom.xml test file at the end of this test. Perhaps with try/finally block.

Reply:
The `TemoporaryFolder` rule being used here deletes the folder and all its contents after the test ends. 

https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html